qqmlsortfilterproxymodel.cpp

```diff
@@ -436,8 +462,9 @@ void QQmlSortFilterProxyModel::updateRoleNames()
     auto roles = m_roleNames.keys();
     auto maxIt = std::max_element(roles.cbegin(), roles.cend());
     int maxRole = maxIt != roles.cend() ? *maxIt : -1;
-    for (auto proxyRole : m_proxyRoles) {
-        for (auto roleName : proxyRole->names()) {
+    for (auto proxyRole : qAsConst(m_proxyRoles)) {
```

Comment:
probably reference could be used here as well

Reply:
![image](https://github.com/status-im/SortFilterProxyModel/assets/5377645/ee4e477f-c382-459b-a6d4-12bca2cdab17)

It's a pointer :)